lib/index.js

```diff
@@ -808,6 +810,58 @@ GA.prototype.productClickedEnhanced = function(track) {
 };
 
 /**
+ * Add Impression - Enhanced Ecommerce
+ *
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#product-impression
+ *
+ * @api private
+ * @param {Track} track
+ */
+
+GA.prototype.productListViewedEnhanced = function(track) {
+  var props = track.properties(); 
+  var products = track.products();
+  this.loadEnhancedEcommerce(track);
+  each(products, function(product) {
+    window.ga('ec:addImpression', {
+      id: product.product_id,
+      category: track.category(),
+      list: props.list_id,
+      position: products.map(function(x) { return x.product_id; }).indexOf(product.product_id)
+    });
+  });
+  window.ga('send', 'pageview'); 
+  this.pushEnhancedEcommerce(track);
+};
+
+/**
+ * Add Impression - Enhanced Ecommerce
+ *
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#product-impression
+ *
+ * @api private
+ * @param {Track} track
+ */
+
+GA.prototype.productListFilteredEnhanced = function(track) {
+  var props = track.properties();
+  var products = track.products();
+  var filters = props.filters.map(function(obj) { return obj.type + ':' + obj.value;}).join();
+  var sorts = props.sorts.map(function(obj) { return obj.type + ':' + obj.value;}).join();
+  this.loadEnhancedEcommerce(track);
+  each(products, function(product) {
+    window.ga('ec:addImpression', {
+      id: product.product_id,
```

Comment:
Similar to above with additional fields:

id: product.product_id || product.sku,
name: product.name,
category: product.category || track.category(),
list: props.list_id || track.category() || 'products',
brand: products.brand,
variant: filters + "::" + sorts, //This seems like a fine usage to get to send filtering and sorting data!
price: products.price,
position: products.map(function(x) { return x.product_id; }).indexOf(product.product_id)